exercises/binary_converter.cpp

```diff
@@ -5,3 +5,38 @@
   Insert first number: 8
   The binary number is: 1000
 */
+
+#include <iostream>
+#include <cmath>
+
+using namespace std;
+
+string convert_to_binary(int number, int bits)
+{
+    string binary(bits, '0');
+    
+    for (int i = 0; i < binary.length(); i++)
+    {
+        if (number > 0)
+        {
+            binary[bits - i - 1] = (char)(number % 2) + 48;
+            number /= 2;
+        }
+    }
+
+    return binary;
+}
+
+int main()
+{
+    int n, bits;
+    do {
+        cout << "Insert number of bits to use: ";
+        cin >> bits;
+    } while (bits < 0 || bits > 16);
+
+    cout << "Insert first number (0-" << pow(2, bits) - 1 << "): ";
+    cin >> n;
+
+    cout << "The binary number is: " << convert_to_binary(n, bits) << endl;
+}
```

Comment:
`int main()` è una funzione che ritorna un valore `int`, è necessario farlo sempre, scrivi quindi `return 0;` prima della chiusura del main(){}

Reply:
Ho capito che è il mio tasto dolente questo ahaha

Reply:
capita sempre, comunque il "return 0" indica che il processo è andato a buon fine

è contro-intuitivo perchè normalmente "0" indica "false" e "1" indica "true", ma nei processi è al contrario

anche se il compilatore è buono e ti accetta il codice così com'è, in realtà sotto banco ti aggiunge quel return 0, perchè è obbligatorio

forse le ultime versioni di g++ ti ritornano errore se non metti il return